cpp/autosar/test/rules/A8-4-8/test.cpp

```diff
@@ -37,3 +37,38 @@ void f7(A &a) { // NON_COMPLIANT
 void f8(int i) { // COMPLIANT
   i += 1;
 }
+
+constexpr A &operator|=(
+    A &lhs,
+    const A &rhs) noexcept { // COMPLIANT - non-member user defined assignment
+                             // operators are considered an exception.
+  return lhs;
+}
+
+enum class byte : unsigned char {};
+constexpr byte &operator|(const byte &lhs, const byte &rhs) noexcept {
```

Comment:
label as compliant, 
also I am not immediately sure what this tests other than that the rule correctly defines out parameters , maybe double check if it contributes something unique, and if so the compliant label with a note will help that